### PR TITLE
Bugfix: don't add errors indefinitely when wagmi hook fails

### PR DESCRIPTION
**Pull request by southbears**
Task details: https://app.dework.xyz/nation3?taskId=4dddb358-a106-4d86-a0bd-3619dbfa5aab

This was a bug that I probably introduced here: https://github.com/nation3/app/pull/67/files#diff-a37b142e8f2323627230313ed4b7932bdf22bc797e0f33bb7cdbb98bff1a5e0cR11

Right now lots of stuff in the app isn't memoized properly, and since a lot of logic is `useEffect` driven, these kinds of bugs will appear. Ideally everything should be memoized, and all deps should be passed to e.g. `useCallback` and `useEffect`, and they won't trigger indefinitely because they're memoized. In this case, we fixed the lint warnings (which tell you to add all deps a hook uses), but that caused this issue because the stuff we put into the deps was not memoized and thus changed on each render.

Ofc more than happy to share more about best practices around memoization and TS I've learned over the years. Also, I'm more than happy to not get the bounty for this one since it was introduced by me in the other PR

### CODE REVIEW

File: ui/components/ErrorProvider.tsx [hunk 1]
```diff
@@ -33,11 +32,11 @@ function ErrorProvider({ children }: any) {
             return
           }
         }
-        setErrors([{ key: count, ...error }, ...errors])
-        setCount(++count)
+        setErrors((prev) => [...prev, {key: prev.length, ...error}]);
```
Comment (gregfromstl) [inline ui/components/ErrorProvider.tsx:35]: Could we remove the `count` state based on using `length` instead?

**Review by gregfromstl — APPROVED** [review]
Looks good besides the ability to remove `count`. Appreciate your expertise on this!

**Comment by southbears** [comment]
@gregfromstl done!

File: ui/lib/use-handle-error.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
-import { useEffect } from 'react'
-import { useErrorContext } from '../components/ErrorProvider'
+import { useEffect } from 'react';
+import { useErrorContext } from '../components/ErrorProvider';
```
Comment (aahna-ashina) [inline ui/lib/use-handle-error.ts:2]:
```suggestion
import { useEffect } from 'react'
import { useErrorContext } from '../components/ErrorProvider'
```

**Review by aahna-ashina — APPROVED** [review]
✅ Tested MetaMask rejects on https://app-62i3nugm4-nation3.vercel.app/lock and https://app-62i3nugm4-nation3.vercel.app/liquidity

**Pull request merged**